kong/api/routes/debug.lua

```diff
@@ -0,0 +1,105 @@
+local get_sys_filter_level = require("ngx.errlog").get_sys_filter_level
+local set_log_level = require("resty.kong.log").set_log_level
+
+local LOG_LEVELS = require("kong.constants").LOG_LEVELS
+
+local ngx = ngx
+local kong = kong
+local pcall = pcall
+local type = type
+local tostring = tostring
+
+local NODE_LEVEL_BROADCAST = false
+local CLUSTER_LEVEL_BROADCAST = true
+
+local function handle_put_log_level(self, broadcast)
+  if kong.configuration.database == "off" then
```

Comment:
From the Design Doc, section "Phase 2 scope (future work)":

> Currently it’s not possible to dynamically change the log level on DB-less nodes because one of the must-have product requirements for the dynamic log level API is that it needs to be protected by RBAC and RBAC is not available in DB-less. Unix domain sockets or local ports is not an acceptable solution because if running a CLI like kong set-log-level is the only option, Kubernetes users can't easily run such CLI in a production Kubernetes cluster.

Please refer to FT-3550 from the Phase 2 Epic, there has been some suggestions in Phase 2 to introduce a new `debug_listen` for DB-less nodes and expose this API to that listener. Add your ideas about it there.